.github/workflows/coverage.yml

```diff
@@ -0,0 +1,60 @@
+name: Generate code coverage badge
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Update coverage badge
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal access token.
+          fetch-depth: 0 # otherwise, there would be errors pushing refs to the destination repository.
+
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.20'
```

Comment:
this is fine for now but with setup-go@v3 they support a "stable" alias which would lower toil of maintaining this over time.

Reply:
also setup-go v4 does some automatic caching I think (we can add this later)